egs/wsj/s5/steps/cleanup/append_eval_to_ctm.py

```diff
@@ -0,0 +1,396 @@
+#!/usr/bin/env python
+
+# Copyright 2016  Vimal Manohar
+# Apache 2.0
+
+from __future__ import print_function
+import sys, operator, argparse
+
+# Append to the CTM the Levenshtein alignment of 'hypothesis' and 'reference':
+# (i.e. the output of 'align-text' post-processed by 'wer_per_utt_details.pl')
+
+# The tags in the processed Levenshtein alignment are:
+#  'C' = correct
+#  'S' = substitution
+#  'D' = deletion
+#  'I' = insertion
+#  'EC' = silence and neighboring words are both correct
+#  'ES' = silence and one of the neighboring words is a substitution
+#  'ED' = silence and one of the neighboring words is a deletion
+#  'EI' = silence and one of the neighboring words is an insertion
+#  'XS' = substitution by an OOV
+
+def enum(*sequential, **named):
+    enums = dict(zip(sequential, range(len(sequential))), **named)
+    reverse = dict((value, key) for key, value in enums.iteritems())
+    enums['reverse_mapping'] = reverse
+    return type('Enum', (), enums)
+
+def GetArgs():
+    parser = argparse.ArgumentParser(description =
+        """Append to the CTM the Levenshtein alignment of 'hypothesis' and 'reference' :
+        (i.e. the output of 'align-text' post-processed by 'wer_per_utt_details.pl')""",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument("--special-symbol", default = "***",
+                        help = "Special symbol used to align insertion or deletion "
+                        "in align-text binary")
+    parser.add_argument("--silence-symbol", default = "<eps>",
+                        help = "Must be provided to ignore silence words in the "
+                        "CTM that would be present if --print-silence was true in "
+                        "nbest-to-ctm binary")
+    parser.add_argument("--oov-symbol", default = "<unk>",
+                        help = "Substitutions by oov are treated specially")
+
+    # Required arguments
+    parser.add_argument("eval_in", metavar = "<eval-in>",
+                        help = "Output of 'align-text' post-processed by 'wer_per_utt_details.pl'")
+    parser.add_argument("ctm_in", metavar = "<ctm-in>",
+                        help = "Hypothesized CTM")
+    parser.add_argument("ctm_eval_out", metavar = "<ctm-eval-out>",
+                        help = "CTM appended with word-edit information. ")
+    args = parser.parse_args()
+
+    return args
+
+def CheckArgs(args):
+    args.ctm_eval_out_handle = open(args.ctm_eval_out, 'w')
+
+    if args.silence_symbol == args.special_symbol:
+        print("WARNING: --silence-symbol and --special-symbol are the same", file = sys.stderr)
+
+    return args
+
+EditType = enum('C', 'S', 'D', 'I', 'EC', 'ES', 'ED', 'EI', 'XS')
```

Comment:
this enum is not used in the code- you might want to remove it or make it a comment.  I notice it appears in two scripts.